File: frontend/src/microservices/advertising/objects/GenerateAdvertisementTextCandidatesRequest.ts
// 本文件定义 advertising 模块的 `GenerateAdvertisementTextCandidatesRequest`，作为请求参数并提供 JSON 编解码。

export type GenerateAdvertisementTextCandidatesRequest = {
  prompt: string
  sourceText?: string | null
  styleRequirement?: string | null
  focus?: string | null
  tone?: string | null
  resourceLabel?: string | null
  advertisementKind?: string | null
  candidateCount?: number | null
  avoidText?: string | null
}

export const generateAdvertisementTextCandidatesRequestFromJson = (json: string): GenerateAdvertisementTextCandidatesRequest =>
  JSON.parse(json) as GenerateAdvertisementTextCandidatesRequest

export const generateAdvertisementTextCandidatesRequestToJson = (value: GenerateAdvertisementTextCandidatesRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/GenerateAdvertisementTextCandidatesRequest.scala
// 本文件定义 advertising 模块的 GenerateAdvertisementTextCandidatesRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GenerateAdvertisementTextCandidatesRequest(
    prompt: String,
    sourceText: Option[String],
    styleRequirement: Option[String],
    focus: Option[String],
    tone: Option[String],
    resourceLabel: Option[String],
    advertisementKind: Option[String],
    candidateCount: Option[Int],
    avoidText: Option[String]
)

object GenerateAdvertisementTextCandidatesRequest:
  given sourceEncoder: Encoder[GenerateAdvertisementTextCandidatesRequest] = deriveEncoder
  given sourceDecoder: Decoder[GenerateAdvertisementTextCandidatesRequest] = deriveDecoder

